benchmark/benchmark_tridiagonal_solver.cpp

```diff
@@ -0,0 +1,108 @@
+#include "mkl_lapacke.h"
```

Comment:
we could add the `#include <lapacke.h>` back for GNU. I tested it and it worked for gcc/12 on Derecho.